src/main/java/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionService.java

```diff
@@ -72,7 +82,21 @@ public TranslatedNonMatchingResponseBody translateSuccessResponse(List<Assertion
 
         validate(authnAssertion, mdsAssertion, expectedInResponseTo, expectedLevelOfAssurance, levelOfAssurance);
 
+        validate(authnAssertion, mdsAssertion, expectedInResponseTo, expectedLevelOfAssurance, levelOfAssurance);
+
+        AuthnStatement authnStatement = authnAssertion.getAuthnStatements().get(0);
         String nameID = mdsAssertion.getSubject().getNameID().getValue();
+        String issuerId =  authnAssertion.getIssuer().getValue();
+
+        String hashId = userIdHashFactory.hashId(
+                issuerId,
```

Comment:
I'd like to check with Andy, but I *think* this is wrong.  The issuerId here is the IDP's ID, whereas I think we want the RP's issuer ID.

Reply:
You may be right;  I have just implement this just as was done in MSA. It seems the idea of the above is that a different hashid depending on the selected IDP

Reply:
Actually, you're right.  We do need to use the IDP's issuer ID, and the RP's issuer ID is already included, by virtue of it being passed into the UserIdHashFactory when it is created.  So, what you've got here is fine; sorry for muddying the waters!